mpcs/src/basefold.rs

```diff
@@ -213,91 +206,90 @@ where
 
     fn batch_commit(
         pp: &Self::ProverParam,
-        rmm: witness::RowMajorMatrix<<E as ff_ext::ExtensionField>::BaseField>,
+        rmms: BTreeMap<usize, witness::RowMajorMatrix<<E as ff_ext::ExtensionField>::BaseField>>,
     ) -> Result<Self::CommitmentWithWitness, Error> {
-        let span = entered_span!("to_mles", profiling_3 = true);
-        let polys = rmm.to_mles();
-        exit_span!(span);
-        // assumptions
-        // 1. there must be at least one polynomial
-        // 2. all polynomials must exist in the same field type
-        //    (TODO: eliminate this assumption by supporting commiting
-        //     and opening mixed-type polys)
-        // 3. all polynomials must have the same number of variables
-
-        if polys.is_empty() {
+        if rmms.is_empty() {
             return Err(Error::InvalidPcsParam(
                 "cannot batch commit to zero polynomials".to_string(),
             ));
         }
 
-        let num_vars = polys[0].num_vars();
-        let num_polys = polys.len();
+        let mmcs = poseidon2_merkle_tree::<E>();
 
-        let is_base = match polys[0].evaluations() {
-            FieldType::Ext(_) => false,
-            FieldType::Base(_) => true,
-            _ => unreachable!(),
-        };
+        let span = entered_span!("to_mles", profiling_3 = true);
+        let (polys, rmm_to_batch_commit, trivial_proofdata, circuit_codeword_index): (
```

Comment:
nitpick: shall we rename `trivial_proofdata` to `rmms_without_encoding` or add some comment to make it clear?

Reply:
yeah it make sense. comment added in BatchCommitmentWithWitness structure https://github.com/scroll-tech/ceno/pull/894/commits/d957832a01215de6cf8e57b02623e66ae18dcc47